plugins/redfish/fu-redfish-legacy-device.c

```diff
@@ -17,9 +18,40 @@ struct _FuRedfishLegacyDevice {
 
 G_DEFINE_TYPE(FuRedfishLegacyDevice, fu_redfish_legacy_device, FU_TYPE_REDFISH_DEVICE)
 
+typedef struct curl_slist _curl_slist;
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(_curl_slist, curl_slist_free_all)
+
+G_DEFINE_AUTOPTR_CLEANUP_FUNC(curl_mime, curl_mime_free)
+
+/* This function will determine if an attach or detach action will be run
+ * on the device. This is useful for devices that don't want to be detached
+ * or attached, like HPE servers.
+ *
+ * inputs:
+ * - dev: the device to check
+ * - action: 0 for attach, 1 for detach
```

Comment:
Mmm I don't really like magic numbers without an enum for arguments.  Either make this an enum or make it a gboolean `detach`